itest/lnd_channel_backup_test.go

```diff
@@ -1672,3 +1954,383 @@ func assertDLPExecuted(ht *lntest.HarnessTest,
+	// Carol should force close the channel using her latest commitment.
+	// For scbforceclose case: we detect the manually broadcasts Dave's tx.
+	ht.AssertNumTxsInMempool(1)
+
+	// Dave should also consider the channel "waiting close", as he noticed
+	// the channel was out of sync, and is now waiting for a force close to
+	// hit the chain.
+	ht.AssertNumWaitingClose(dave, 1)
+
+	// Restart Dave to make sure he is able to sweep the funds after
+	// shutdown.
+	ht.RestartNode(dave)
+
+	// Generate a single block, which should confirm the closing tx.
+	ht.MineBlocksAndAssertNumTxes(1, 1)
+	blocksMined := uint32(1)
+
+	// Start Carol after mining closing tx if requested. It is done here to
+	// ensure that she does not publish her version of force close tx.
+	require.NoError(ht, startCarol())
+	ht.EnsureConnected(carol, dave)
+
+	// Dave should consider the channel pending force close (since he is
+	// waiting for his sweep to confirm).
+	ht.AssertNumPendingForceClose(dave, 1)
+
+	// Carol is considering it "pending force close", as we must
+	// wait before she can sweep her outputs.
+	ht.AssertNumPendingForceClose(carol, 1)
+
+	if commitType == lnrpc.CommitmentType_SCRIPT_ENFORCED_LEASE {
+		// Dave should sweep his anchor only, since he still has the
+		// lease CLTV constraint on his commitment output. We'd also
+		// see Carol's anchor sweep here.
+
+		// Both Dave and Carol should have an anchor sweep request.
+		// Note that they cannot sweep them as these anchor sweepings
+		// are uneconomical.
+		ht.AssertNumPendingSweeps(dave, 1)
+
+		// Carol has two sweeps.
+		ht.AssertNumPendingSweeps(carol, 2)
+
+		// After Carol's output matures, she should also reclaim her
+		// funds.
+		//
+		// The commit sweep resolver publishes the sweep tx at
+		// defaultCSV-1 and we already mined one block after the
+		// commitmment was published, so take that into account.
+		ht.MineEmptyBlocks(int(defaultCSV - blocksMined))
+
+		// Carol should have two sweep requests - one for her commit
+		// output and the other for her anchor.
+		ht.AssertNumPendingSweeps(carol, 2)
+
+		// Mine a block to trigger the sweep in Carol.
+		ht.MineEmptyBlocks(1)
+		ht.MineBlocksAndAssertNumTxes(1, 1)
+
+		// Now the channel should be fully closed also from
+		// Carol's POV.
+		ht.AssertNumPendingForceClose(carol, 0)
+
+		// We'll now mine the remaining blocks to prompt Dave to sweep
+		// his CLTV-constrained output.
+		resp := dave.RPC.PendingChannels()
+		blocksTilMaturity :=
+			resp.PendingForceClosingChannels[0].BlocksTilMaturity
+		require.Positive(ht, blocksTilMaturity)
+
+		ht.MineEmptyBlocks(int(blocksTilMaturity))
+
+		// Dave should have two sweep requests - one for his commit
+		// output and the other for his anchor.
+		ht.AssertNumPendingSweeps(dave, 2)
+
+		// Mine a block to trigger the sweep.
+		ht.MineEmptyBlocks(1)
+		ht.MineBlocksAndAssertNumTxes(1, 1)
+
+		// Now Dave should consider the channel fully closed.
+		ht.AssertNumPendingForceClose(dave, 0)
```

Comment:
I think there is no need to test the sweeper here again imo, checking whether the sweeps where requested is sufficient enough.

Reply:
Fixed. Removed extra checks.